### ISSUE TITLE
match expr with optional propagation occurs C error

### ISSUE THREAD

**Issue by zakuro9715** [vlang/v#9556]
**V version:**  V 0.2.2 043f642
**OS:**  linux, Ubuntu 18.04.5 LTS (WSL 2)

**What did you do?**

```
fn g() ?int {
        return 0
}
fn f(v int) ?int {
        return match v {
                0 { g()? }
                1 { g()? }
                else { g()? }
        }
}

println(f(0))
```

**What did you expect to see?**

it can be compiled

**What did you see instead?**

```
/tmp/v/main.6062899099655765079.tmp.c:9227:11: error: unexpected type name 'Option_int': expected expression
                _t103 = Option_int _t104 = main__g();
                        ^
/tmp/v/main.6062899099655765079.tmp.c:9228:7: error: use of undeclared identifier '_t104'
                if (_t104.state != 0) { /*or block*/
                    ^
/tmp/v/main.6062899099655765079.tmp.c:9230:20: error: use of undeclared identifier '_t104'
                        memcpy(&_t105, &_t104, sizeof(Option));
                                        ^
/tmp/v/main.6062899099655765079.tmp.c:9235:11: error: unexpected type name 'Option_int': expected expression
                _t103 = Option_int _t106 = main__g();
                        ^
/tmp/v/main.6062899099655765079.tmp.c:9236:7: error: use of undeclared identifier '_t106'
                if (_t106.state != 0) { /*or block*/
                    ^
/tmp/v/main.6062899099655765079.tmp.c:9238:20: error: use of undeclared identifier '_t106'
                        memcpy(&_t107, &_t106, sizeof(Option));
                                        ^
/tmp/v/main.6062899099655765079.tmp.c:9243:11: error: unexpected type name 'Option_int': expected expression
                _t103 = Option_int _t108 = main__g();
                        ^
/tmp/v/main.6062899099655765079.tmp.c:9244:7: error: use of undeclared identifier '_t108'
                if (_t108.state != 0) { /*or block*/
                    ^
/tmp/v/main.6062899099655765079.tmp.c:9246:20: error: use of undeclared identifier '_t108'
                        memcpy(&_t109, &_t108, sizeof(Option));
                                        ^

```

generated c code is:

```
Option_int main__f(int v) {
	int _t103;
	
	if (v == (0)) {
		_t103 = Option_int _t104 = main__g();
		if (_t104.state != 0) { /*or block*/ 
			Option_int _t105;
			memcpy(&_t105, &_t104, sizeof(Option));
			return _t105;
		};
	}
	else if (v == (1)) {
		_t103 = Option_int _t106 = main__g();
		if (_t106.state != 0) { /*or block*/ 
			Option_int _t107;
			memcpy(&_t107, &_t106, sizeof(Option));
			return _t107;
		};
	}
	else {
		_t103 = Option_int _t108 = main__g();
		if (_t108.state != 0) { /*or block*/ 
			Option_int _t109;
			memcpy(&_t109, &_t108, sizeof(Option));
			return _t109;
		};
	}	Option_int _t102;
	opt_ok(&(int[]) { _t103 }, (Option*)(&_t102), sizeof(int));
	return _t102;
}


```

**Comment by JalonSolov** [comment]
Still not fixed, but the error message is much shorter with `V 0.2.2 cbf30bd`
```
/tmp/v/foo.8323988203407798405.tmp.c:9508: error: cannot convert 'typedef struct Option_int' to 'int'
...
```

**Comment by Dialga** [comment]
Fixed by #12032.
